### PR TITLE
Added riptide-problem and riptide-exceptions

### PR DESCRIPTION
**Pull request by whiskeysierra**

### CODE REVIEW

**Comment by whiskeysierra** [comment]
> Cool stuff, should we add a small paragraph each to the README? Maybe even a table is sufficient.

Absolutely. I'm not sure whether each module deserves its own README or just a section in the main. I'd like the keep the big one as small as possible.

File: riptide-exceptions/src/main/java/org/zalando/riptide/exceptions/TemporaryExceptionPlugin.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import static org.zalando.fauxpas.FauxPas.rethrow;
+import static org.zalando.fauxpas.FauxPas.throwingFunction;
+
+public final class TemporaryExceptionPlugin implements Plugin {
+
+    private final Predicate<Throwable> temporary;
+
+    public TemporaryExceptionPlugin() {
+        this(InterruptedIOException.class::isInstance,
```
Comment (jhorstmann) [inline riptide-exceptions/src/main/java/org/zalando/riptide/exceptions/TemporaryExceptionPlugin.java]: IOExceptions in general are not retryable?

Comment (whiskeysierra) [inline riptide-exceptions/src/main/java/org/zalando/riptide/exceptions/TemporaryExceptionPlugin.java]: No, at least not by default. There is just too much stuff that is an IOException that is a permanent error:

- CharacterCodingException
- CharConversionException
- FileNotFoundException
- InvalidPropertiesFormatException
- MalformedURLException
- UnsupportedEncodingException

To name a few (just from the JDK alone).

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**